x/gov/keeper.go

```diff
@@ -382,7 +377,7 @@ func (keeper Keeper) AddDeposit(ctx sdk.Context, proposalID uint64, depositorAdd
 	}
 
 	// Send coins from depositor's account to DepositedCoinsAccAddr account
-	_, err := keeper.ck.SendCoins(ctx, depositorAddr, DepositedCoinsAccAddr, depositAmount)
+	_, _, err := keeper.ck.SubtractCoins(ctx, depositorAddr, depositAmount)
```

Comment:
I prefer that we have the DepositedCoinsAccAddr... it's weird that a deposit would change an invariant of the total number of coins.

Reply:
It's also useful to be able to query for the total number of deposited coins, etc.
+ calling subtract/add is dangerous... e.g. a bug in this code could result in hyperinflation.  The Send method ensures that only a limit is sent.

Reply:
If you're concerned about the hash collision, we have bigger problems if there's a hash collision.  It's strictly OK to create human-readable addresses, we also bumped up the []byte to 32, so lets make use of that.

Reply:
I don't think than an account is not the right place to be storing this data.  The sum total of all deposits isn't "an account".  One suggestion we had instead was that we could have a function that all keepers expose that allow them to declare how many coins they have under their control.  That way we could still have an invariant that checks the total supply is still correct.

`AddCoins` and `SubtractCoins` can't change the coin's supply, only `MintAddCoins` and `BurnSubtractCoins` can.

And yeah, not worried about a hash collision, but just the general abstraction here.